cmd/build/helmify/kustomization.yaml

```diff
@@ -78,3 +78,9 @@ patchesJson6902:
         path: /spec/template/spec/nodeSelector/kubernetes.io~1os
       - op: remove
         path: /spec/template/spec/affinity/podAntiAffinity
+      - op: remove
```

Comment:
Thanks @maxsmythe for the tip about the ", that did the trick. I guess I thought k8s would create some issues with a initial `:` without a " around it but apparently not :) 

About the remove stuff, I'm not to happy about it ether.
I have tried to write a path that would solve it but my kustomize skills is not enough within this area.

I have tried a number of different options, these are two of them: 

```.yaml
      - op: replace
        path: /spec/template/spec/containers/0/ports/8888
        value: HELMSUBST_DEPLOYMENT_CONTROLLER_MANAGER_PORT
```

```.yaml
      - op: replace
        path: /spec/template/spec/containers/0/ports/containerPort/8888
        value: 1234
```

And when I run `make manifests` i get:

`Error: replace operation does not apply: doc is missing path: /spec/template/spec/containers/0/ports/containerPort/8888: missing value`

I feel that I'm missing something obvious but I can't find any good example.
Do you have any example of this? Hmm starting to feel like you are writing the PR and not me, sorry about that.

Reply:
@maxsmythe do you have some feedback on this?
I can't come up with any better way of solving this.

Reply:
Sorry for the slow reply.

Here is an example kustomize setup that would remove a container port:

`manager.yaml`:
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: controller-manager
  namespace: system
  labels:
    control-plane: controller-manager
    gatekeeper.sh/operation: webhook
spec:
  selector:
    matchLabels:
      control-plane: controller-manager
      gatekeeper.sh/operation: webhook
  replicas: 3
  template:
    metadata:
      annotations:
        container.seccomp.security.alpha.kubernetes.io/manager: runtime/default
      labels:
        control-plane: controller-manager
        gatekeeper.sh/operation: webhook
    spec:
      affinity:
        podAntiAffinity:
          preferredDuringSchedulingIgnoredDuringExecution:
            - weight: 100
              podAffinityTerm:
                labelSelector:
                  matchExpressions:
                    - key: gatekeeper.sh/operation
                      operator: In
                      values:
                        - webhook
                topologyKey: kubernetes.io/hostname
      serviceAccountName: admin
      automountServiceAccountToken: true
      containers:
        - command:
            - /manager
          args:
            - "--port=8443"
            - "--logtostderr"
            - "--exempt-namespace=gatekeeper-system"
            - "--operation=webhook"
          image: openpolicyagent/gatekeeper:v3.6.0-beta.3
          imagePullPolicy: Always
          name: manager
          ports:
            - containerPort: 8888
              name: metrics
              protocol: TCP
            - containerPort: 9090
              name: healthz
              protocol: TCP
          env:
            - name: POD_NAMESPACE
              valueFrom:
                fieldRef:
                  apiVersion: v1
                  fieldPath: metadata.namespace
            - name: POD_NAME
              valueFrom:
                fieldRef:
                  fieldPath: metadata.name
          resources:
            limits:
              cpu: 1000m
              memory: 512Mi
            requests:
              cpu: 100m
              memory: 256Mi
          livenessProbe:
            httpGet:
              path: /healthz
              port: 9090
          readinessProbe:
            httpGet:
              path: /readyz
              port: 9090
          securityContext:
            readOnlyRootFilesystem: true
            runAsUser: 1000
            runAsGroup: 999
            runAsNonRoot: true
            allowPrivilegeEscalation: false
            capabilities:
              drop:
                - all
      terminationGracePeriodSeconds: 60
      nodeSelector:
        kubernetes.io/os: linux
      priorityClassName: system-cluster-critical
```

`patch.yaml`:
```yaml
kind: Deployment
apiVersion: apps/v1
metadata:
  name: controller-manager
  namespace: system
spec:
  template:
    spec:
      containers:
      - name: manager
        ports:
        - containerPort: 8888
          $patch: delete
```

`kustomization.yaml`:
```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
resources:
  - manager.yaml
patchesStrategicMerge:
  - patch.yaml
```

Reply:
No worries at all, thanks for the great example.
I have seen the `$patch: delete` example earlier but for some reason I thought it would be in the original yaml instead of a patch.,yaml.... Thanks allot!